### ISSUE TITLE
The base model has the same results when add a peft model?

### ISSUE THREAD

**Issue by iopwsy** [huggingface/peft#940]
### System Info

My code is:
`base_model =[]`
`peft_model = []`
`new model = PeftModel.from_pretrained(base_model, peft_model) `
However, if I want to use the base model after these, it has the same results with the new_model unless I restart it.
The only method I use to do is create a zero peft_model.

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder
- [X] My own task or dataset (give details below)

### Reproduction

base_model =[]
peft_model = []
new_model = PeftModel.from_pretrained(base_model, peft_model)
res1 = new_model(**input)
res2 = base_model(**input)

### Expected behavior

`res1` is the same as `res2` but I don't want it

**Comment by BenjaminBossan** [comment]
PEFT will modify the base model in-place to add the adapters, therefore, the base model will output different results from before applying PEFT. If you need the original output at the same time, you may have to load a separate copy of the base model.

**Comment by BenjaminBossan** [comment]
I agree that it can be confusing. The solution you suggested would also be confusing, though, because the `base_model` and the `(peft_)model` are not identical. The former is the base model with some in-place modification, the latter is an instance of `PeftModel` that wraps the `base_model` and adds some stuff on top. If there was a way to implement PEFT without modifying the base model, that would be ideal, but there are a bunch of technical reasons that make it impossible, or at least extremely hard.

**Comment by MaveriQ** [comment]
So what I meant was that we just have a function `convert_to_peft_model` which doesn't return anything. Then it will be less confusing.  

However I am getting another problem. I do a deepcopy of the base_model before applying `get_peft_model` to it. But I get the exact same forward pass result. Here is the code : 

```
config = peft.LoraConfig(
    r=8,
    target_modules=["seq.0", "seq.2"],
    modules_to_save=["seq.4"],
)

base_model = MLP()
orig_model = copy.deepcopy(base_model)
peft_model = peft.get_peft_model(base_model, config)

X = torch.rand((100, 20))
y = (X.sum(1) > 10).long()
criterion = nn.CrossEntropyLoss()

def step(model,xb,yb):
    outputs = model(xb)
    loss = criterion(outputs, yb)
    return loss

step(peft_model,X,y) == step(orig_model,X,y)
```

MLP() is from https://github.com/huggingface/peft/blob/main/examples/multilayer_perceptron/multilayer_perceptron_lora.ipynb

**Comment by BenjaminBossan** [comment]
> So what I meant was that we just have a function `convert_to_peft_model` which doesn't return anything. Then it will be less confusing.

There is such a function, [`inject_adapter_in_model`](https://huggingface.co/docs/peft/developer_guides/low_level_api#injectadapterinmodel-method).

> However I am getting another problem. I do a deepcopy of the base_model before applying `get_peft_model` to it. But I get the exact same forward pass result. Here is the code :

The issue with that code is that by default, applying LoRA does not change the output (some weights are initialized as 0's). If you pass `init_lora_weights=False` to the config, the results will differ. 

Also, in some very rare circumstances, `copy` and PyTorch tensors may have strange interactions. To be 110% certain, I would actually initialize the original model (with the same seed) instead of creating a `deepcopy`.

**Comment by MaveriQ** [comment]
Thank you for the clarification. I hope this discussion will be useful for anyone who faces similar issues.

**Comment by iopwsy** [comment]
> @iopwsy If this answers your initial question, feel free to close the issue.

Thanks a lot. I got it
